### ISSUE TITLE
Define a standard for file/folder structure

### ISSUE THREAD

**Issue by Floppy** [Floppy/van_dam#460]
So that one day we can auto-organise the files on disk, we first need a standard, or a few options for standards.

We need to decide:

* [ ] How to name model folders
* [ ] How to name part files
* [ ] How to structure folders within a model
* [ ] How to structure folders within a library, that contain models 

I would love to get some examples from the community on how they structure their folders.

**Comment by Floppy** [comment]
Probably the right thing to do is have a template string for defining the paths. So you would have a string for "model folder name" which could be something like `{collections}/{name} - {creator}`, and a string for part names, like `{name}.{type}`.

That way you can do things like `{creator}/{name}` to organise by who made the thing, or `{creator}/{tags}/{name}`, and so on.

**Comment by Floppy** [comment]
Fields could be:

* `{model_title}`: The title of the model
* `{part_title}`: The title of the part
* `{supported}`: Supported or not?
* `{collections}`: A folder structure reflecting nested collection  names
* `{creator}`: The creator name
* `{format}`: The file format (3mf, stl, obj)

Global options:

* `separator`: the character used as a separator between words
* `case`: whether path components should be lowercased, titlecased, left alone, etc.

**Comment by Floppy** [comment]
Here's how Sonarr does it:

![Screenshot 2022-03-31 at 22 33 37](https://user-images.githubusercontent.com/3565/161153030-9773dead-4bd7-4de6-ade5-d7f967ed7feb.png)

**Comment by Code-Slave** [comment]
definitely use a configurable format.

**Comment by Floppy** [comment]
So, part of this is done. Currently files can be organised to match a fixed format, which is something like `{tag/tag/tag}/{model-name}#{model-id}`. Next step would be to make this configurable per-library, as that's not always going to be what you want.

**Comment by ksuquix** [comment]
One thing to consider as you are working on this part is the interaction with the scanner.   Say you have a structure like: 
   `{creator}/{collection}/{model_name}`
Then if you add a new model in a new directory under that collection, the scanner should be able to automatically assign the creator and collection.

**Comment by Floppy** [comment]
@ksuquix mm, good point!

**Comment by javijuji** [comment]
Will the readme be updated with folder structure information? Dropping my whole stl file library into the /library makes a big mess right now and very hard to navigate.

**Comment by Floppy** [comment]
At the moment it shouldn't be moving anything around unless you hit "organise files", and there's only one format based on tags. Once it's customisable properly it'll definitely get added to the docs, yes.

**Comment by Netrecov** [comment]
i tend to organinze my library by folders, i use them as catageories, and any sub folder within the parent is for depper classification, ie need to adjust, need to print, different versions etc,

**Comment by ChrisWeiss** [comment]
To add another example - this is MediaMonkey (a Windows app for managing audio files). It's got a fairly powerful tag-from-filename functionality. https://www.mediamonkey.com/wiki/Auto-Tag_from_Filename
<img width="1029" alt="mediamonkey_tag_from_filename" src="https://user-images.githubusercontent.com/419818/221372401-03932204-d27b-4cfa-bd87-445a422d8cb4.png">

**Comment by Floppy** [comment]
This is Sonarr's approach to file naming:

<img width="1055" alt="Screenshot 2023-02-28 at 20 24 14" src="https://user-images.githubusercontent.com/3565/221970249-c04569e0-4fbf-4a83-95d2-7f652903be13.png">

**Comment by Floppy** [comment]
I think maybe having a set of tokens, plus some options for "separator" and "safe filenames" might be good. Though perhaps those latter ones should only be added when actually necessary.

**Comment by DougTheDruid** [comment]
I would say the current structuring is probably the biggest limiter to myself (and probably a lot of others) to really starting to use this software. I have a bunch of Patreons I support and a lot of them follow a similar folder structure, but very rarely is there a consistent naming format.

I would _love_ to simply be able to view my models in a folder structure in van dam, basically the same as if you are using windows explorer w/ preview for images, but you have all of the models for that folder viewable. I think this would give folks a lot of freedom to structure things how they want without needed any super complex filename parsing logic. Bonus points if there was a way to interpret the pre-rendered img files and use them to "preview" whats in a given folder. Here is a horrible inspect-element mock-up of what it might look like: 

![Image](https://user-images.githubusercontent.com/51171051/222532843-42be0534-d635-437f-bc04-8331855adf4e.png)
Basically a new tab in which it will show you all of the content under your "libraries" folder. Folders are shown as a default folder image and open-able just like a normal file explorer. Like mentioned above, you could build in logic to interpret X image name in a directory to represent the parent directory (e.g. a logo for a creator folder, or a pre-rendered image for a release/object folder), then the explorer will only show folders and stl files. Stl files would be shown similarly to today, where you can open them, download, etc.

**Comment by Floppy** [comment]
@DougTheDruid thanks for the idea! I'm only just really learning how people want to use the software, so all this really helps! Could I get you to add your comment as a new issue instead? That will help me organise it more easily and not let it get lost inside another issue, I think it's important enough to have its own 👍🏻

**Comment by Floppy** [comment]
Ooh actually github has a new button, I'll see if I can do it myself...

**Comment by Floppy** [comment]
Customisable file/folder structure is now released in v0.42.0! 🎉
